riptide-faults/src/test/java/org/zalando/riptide/faults/FaultClassifierTest.java

```diff
@@ -35,6 +35,13 @@ public void shouldNotClassifyGenericSSLHandShakeException() {
         assertNotTransient(new SSLHandshakeException("No hands, no cookies"));
     }
 
+    @Test
+    public void shouldClassifyAsTransientWithNonTransientRootCause() {
```

Comment:
I think we need two additional tests, beside testing root cause match:
1. test that verifies matching throwable itself
2. test that verifies intermediate cause (`new Exception("outer", new Exception("intermediate", new Exception("inner")))`)

Reply:
> test that verifies matching throwable itself

See line 20.